zing_stats.py

```diff
@@ -363,27 +431,52 @@ def get_changes_by_project(changes):
     return changes_by_project
 
 
-def parse_gerrit_change_message(msg):
+def parse_ci_job_comments(msg):
     """
     Parse gerrit change messages that look like CI job messages,
     extracting CI job data and returning as a dict
     """
-
-    # TODO refactor to take a list of patterns for runs and jobs (ultimately
-    # taken from a config parameter/file)
     ci_run_patt = 'Patch Set (?P<num>\d+): Verified(?P<v_score>\S+)\s+Build (?P<status>\S+)\s+(?P<jobs>.+)'  # noqa
+
+    return parse_change_messages(msg['message'], ci_run_patt)
+
+
+def parse_pr_message(msg):
+    """
+    Parse PR messages that look like CI job messages,
+    extracting CI job data and returning as a dict
+    """
+    ci_run_patt = 'Build (?P<status>\S+)\s+(?P<jobs>.+)'  # noqa
+
+    return parse_change_messages(msg['body'], ci_run_patt)
+
+
+def parse_change_messages(message, ci_run_patt):
+    """
+    Parse change messages that look like CI job messages,
+    extracting CI job data and returning as a dict
+    """
+
+    # TODO refactor to take a list of patterns for runs/jobs from a file
     ci_run_re = re.compile(ci_run_patt, re.MULTILINE | re.DOTALL)
     ci_job_v1_patt = '^- (?P<proto>.+)?://(?P<jenkins_path>.+)?/job/(?P<name>\S+)/\d+/ : (?P<result>\S+) in (?P<time_h>\d+h )?(?P<time_m>\d+m )?(?P<time_s>\d+s)(?P<non_voting> \(non\-voting\))?(?P<the_rest>.*)$'  # noqa
     ci_job_v1_re = re.compile(ci_job_v1_patt, re.MULTILINE)
     ci_job_v2_patt = '^- (?P<proto>.+)?://(?P<logs_path>.+)?/(?P<name>\D+) : (?P<result>\S+) in (?P<time_h>\d+h )?(?P<time_m>\d+m )?(?P<time_s>\d+s)(?P<non_voting> \(non\-voting\))?(?P<the_rest>.*)$'  # noqa
     ci_job_v2_re = re.compile(ci_job_v2_patt, re.MULTILINE)
 
     run = dict()
-    ci_run_match = ci_run_re.match(msg['message'])
+    ci_run_match = ci_run_re.match(message)
     if ci_run_match:
-        run['date'] = msg['date']
-        run['num'] = ci_run_match.group('num')
-        run['v_score'] = ci_run_match.group('v_score')
+        try:
+            run['num'] = ci_run_match.group('num')
```

Comment:
This requires knowledge of the pattern the caller is using, same in the call below. Should you do the initial matching in the caller and pass the resulting matcher object and initial dict down to this method, or maybe at least make this a private method so it's clear it shouldn't be called directly outside of this module?

Reply:
Making private for now and adding suggestion as a TODO